controllers/machineset_controller_test.go

```diff
@@ -228,6 +228,10 @@ func TestMachineSetReconciler(t *testing.T) {
 			g.Expect(im.GetAnnotations()).To(HaveKeyWithValue("annotation-1", "true"), "have annotations of MachineTemplate applied")
 			g.Expect(im.GetAnnotations()).To(HaveKeyWithValue("precedence", "MachineSet"), "the annotations from the MachineSpec template to overwrite the infrastructure template ones")
 			g.Expect(im.GetLabels()).To(HaveKeyWithValue("label-1", "true"), "have labels of MachineTemplate applied")
+			g.Expect(im.GetOwnerReferences()).To(HaveLen(2))
+			for _, o := range im.GetOwnerReferences() {
+				g.Expect(o.Kind).To(Or(Equal("Machine"), Equal("MachineSet")))
```

Comment:
nit: we could go farther and check that the Machine owner is the controller e.g `GetControllerOf`

Reply:
Same consideration of https://github.com/kubernetes-sigs/cluster-api/pull/5865#discussion_r769020994 applies.
Verifying that the machine controller applies it own ownerRef is out of scope